### ISSUE TITLE
run-all with not terraform command resuls in panic

### ISSUE THREAD

**Issue by cschroer** [gruntwork-io/terragrunt#1825]
if you run `terragrunt run-all` without any terraform command like this:

`terragrunt run-all --terragrunt-parallelism 10`

terragrunt crashes with some panic:

```
panic: runtime error: index out of range [0] with length 0

goroutine 57 [running]:
github.com/gruntwork-io/terragrunt/options.(*TerragruntOptions).InsertTerraformCliArgs(0xc01174ea80, {0x28a6000, 0xc0041e4fc0, 0x10000000100bda7})
	github.com/gruntwork-io/terragrunt/options/options.go:337 +0x3cf
github.com/gruntwork-io/terragrunt/cli.runTerragruntWithConfig(0xc001f6a1c0, 0xc01174ea80, 0xc018b7d7a0, 0x15)
	github.com/gruntwork-io/terragrunt/cli/cli_app.go:638 +0x7f
github.com/gruntwork-io/terragrunt/cli.RunTerragrunt(0xc001f6a1c0)
	github.com/gruntwork-io/terragrunt/cli/cli_app.go:492 +0x953
github.com/gruntwork-io/terragrunt/configstack.(*runningModule).runNow(0xc00f13f4a0)
	github.com/gruntwork-io/terragrunt/configstack/running_module.go:239 +0x118
github.com/gruntwork-io/terragrunt/configstack.(*runningModule).runModuleWhenReady(0x0, 0xc00018e060)
	github.com/gruntwork-io/terragrunt/configstack/running_module.go:202 +0x9a
github.com/gruntwork-io/terragrunt/configstack.runModules.func1(0x0)
	github.com/gruntwork-io/terragrunt/configstack/running_module.go:172 +0x58
created by github.com/gruntwork-io/terragrunt/configstack.runModules
	github.com/gruntwork-io/terragrunt/configstack/running_module.go:170 +0x95
```
terragrunt version: 0.32.4

There is no terraform arg, so this fails: [`terragruntOptions.TerraformCliArgs[0]`](https://github.com/gruntwork-io/terragrunt/blob/v0.32.4/options/options.go#L337)

**Comment by denis256** [comment]
Hi, I tried to run similar command on some test repositories, however, I get an error about missing arguments:

Example repo:
https://github.com/denis256/terragrunt-test-1499

```
$ terragrunt --version 
terragrunt version v0.32.4

$ terragrunt  run-all --terragrunt-parallelism 10
INFO[0000] Stack at /tmp/terragrunt-test-1499:
  => Module /tmp/terragrunt-test-1499/app (excluded: false, dependencies: [/tmp/terragrunt-test-1499/module1])
  => Module /tmp/terragrunt-test-1499/module1 (excluded: false, dependencies: []) 
Usage: terraform [global options] <subcommand> [args]

The available commands for execution are listed below.
The primary workflow commands are given first, followed by
less common or more advanced commands.

Main commands:
  init          Prepare your working directory for other commands
  validate      Check whether the configuration is valid
  plan          Show changes required by the current configuration
  apply         Create or update infrastructure
  destroy       Destroy previously-created infrastructure

All other commands:
  console       Try Terraform expressions at an interactive command prompt
  fmt           Reformat your configuration in the standard style
  force-unlock  Release a stuck lock on the current workspace
  get           Install or upgrade remote Terraform modules
  graph         Generate a Graphviz graph of the steps in an operation
  import        Associate existing infrastructure with a Terraform resource
  login         Obtain and save credentials for a remote host
  logout        Remove locally-stored credentials for a remote host
  output        Show output values from your root module
  providers     Show the providers required for this configuration
  refresh       Update the state to match remote systems
  show          Show the current state or a saved plan
  state         Advanced state management
  taint         Mark a resource instance as not fully functional
  test          Experimental support for module integration testing
  untaint       Remove the 'tainted' state from a resource instance
  version       Show the current Terraform version
  workspace     Workspace management

Global options (use these before the subcommand, if any):
  -chdir=DIR    Switch to a different working directory before executing the
                given subcommand.
  -help         Show this help output, or the help for a specified subcommand.
  -version      An alias for the "version" subcommand.
ERRO[0000] Module /tmp/terragrunt-test-1499/module1 has finished with an error: 1 error occurred:
        * exit status 127
  prefix=[/tmp/terragrunt-test-1499/module1] 
ERRO[0000] Dependency /tmp/terragrunt-test-1499/module1 of module /tmp/terragrunt-test-1499/app just finished with an error. Module /tmp/terragrunt-test-1499/app will have to return an error too.  prefix=[/tmp/terragrunt-test-1499/app] 
ERRO[0000] Module /tmp/terragrunt-test-1499/app has finished with an error: Cannot process module Module /tmp/terragrunt-test-1499/app (excluded: false, dependencies: [/tmp/terragrunt-test-1499/module1]) because one of its dependencies, Module /tmp/terragrunt-test-1499/module1 (excluded: false, dependencies: []), finished with an error: 1 error occurred:
        * exit status 127
  prefix=[/tmp/terragrunt-test-1499/app] 
ERRO[0000] 2 errors occurred:
        * exit status 127
        * Cannot process module Module /tmp/terragrunt-test-1499/app (excluded: false, dependencies: [/tmp/terragrunt-test-1499/module1]) because one of its dependencies, Module /tmp/terragrunt-test-1499/module1 (excluded: false, dependencies: []), finished with an error: 1 error occurred:
        * exit status 127
```

It is possible to share an example project on which this issue happens?

**Comment by yorinasub17** [comment]
Yup `run-all` needs an arg, but we don't have good error handling for that. We should definitely provide a better error message here, and a PR to fix that would be very welcome!

**Comment by yorinasub17** [comment]
Fixed in #1836
